onnxruntime/python/tools/quantization/README.md

```diff
@@ -0,0 +1,102 @@
+    ```python
+    quantized_model = quantize(model, quantization_mode=QuantizationMode.IntegerOps_Static,
+                               input_quantization_params={
+                                    'input_1': [np.uint8(113), np.float32(0.05)]
+                               })
+    ```
+
+- **QuantizationMode.IntegerOps_Dynamic**:
+    Quantize using integer ops. Only ConvInteger and MatMulInteger ops are supported now.
+    Inputs/activations are quantized using dynamic scale and zero point values which are computed while running the model.
+    ```python
+    quantized_model = quantize(model, quantization_mode=QuantizationMode.IntegerOps_Dynamic)
+    ```
+
+- **QuantizationMode.QLinearOps_Static**:
+    Quantize using QLinear ops. Only [QLinearConv](https://github.com/onnx/onnx/blob/master/docs/Operators.md#qlinearconv) and [QLinearMatMul](https://github.com/onnx/onnx/blob/master/docs/Operators.md#QLinearMatMul) ops are supported now.
+    Inputs/activations are quantized using static scale and zero point values which are specified through "input_quantization_params" option.
+    Output scale and zero point values have to be specified using "output_quantization_params" option.
+    ```python
+    quantized_model = quantize(model, quantization_mode=QuantizationMode.QLinearOps_Static,
+                               input_quantization_params={
+                                    'input_1': [np.uint8(113), np.float32(0.05)]
+                               },
+                               output_quantization_params={
+                                    'output_1': [np.uint8(113), np.float32(0.05)]
+                               })
+    ```
+
+- **QuantizationMode.QLinearOps_Dynamic**:
+    Quantize using QLinear ops. Only QLinearConv and QLinearMatMul ops are supported now.
+    Inputs/activations are quantized using dynamic scale and zero point values which are computed while running the model.
+    Output scale and zero point values have to be specified using "output_quantization_params" option.
+    ```python
+    quantized_model = quantize(model, quantization_mode=QuantizationMode.QLinearOps_Dynamic,
+                               output_quantization_params={
+                                    'output_1': [np.uint8(113), np.float32(0.05)]
+                               })
+    ```
+
+## Options
+
+See below for a description of all the options to quantize():
+
+- **model**: ModelProto to quantize
+- **per_channel**: *default: True*
+	If True, weights of Conv nodes are quantized per output channel. 
+	If False, they are quantized per tensor. Refer [QLinearConv](https://github.com/onnx/onnx/blob/master/docs/Operators.md#qlinearconv) for more information.
+- **nbits**: *default: 8*
+	Number of bits to represent quantized data. Currently only nbits=8 is supported.
+- **quantization_mode**: *default: QuantizationMode.IntegerOps_Dynamic*
+	Can be one of the QuantizationMode values.
+- **asymmetric_input_types**: *default: False*
```

Comment:
I was thinking about these options... I think it would make more sense to change them to
1. VNNI Optimized - In this mode we use u8_s8 conversion with zero point as zero
2. ARM optimized - In this mode we use u8_u8 conversion with zero point not equal to zero

This way users understand better why the input types are different. It is easier to choose the right mode this way. Let me know what you think.

Reply:
> I was thinking about these options... I think it would make more sense to change them to
> 
> VNNI Optimized - In this mode we use u8_s8 conversion with zero point as zero
> ARM optimized - In this mode we use u8_u8 conversion with zero point not equal to zero
> 
> This way users understand better why the input types are different. It is easier to choose the right mode this way. Let me know what you think.

i'm all in favor of making the options more intuitive for users,
but i'm not sure ARM, VNNI optimized is the best choice. In most contexts, ARM isn't something comparable with VNNI and may cause confusion? One could bring up intel vnni, arm in documentation as examples of why to use u8_s8 vs u8_u8 
but it may be preferable to not make any references to hardware/architectures in the option names

Reply:
Another way is to make it execution provider specific nGraph vs CPU for example... however there could be some issues with this approach as well let's talk in detail offline

I do want to change the current options because they are confusing ...

Reply:
Let's keep this the way it is right now... Once we have more clarity on this we can come back and change this.